src/components/DealProposalForm.tsx

```diff
@@ -52,21 +53,37 @@ function Field({
   );
 }
 
+function blockToTime(block: number, currentBlock: number, currentBlockTimestamp: Date): Date {
+  const timeDifference = (block - currentBlock) * BLOCK_TIME;
+  const realTime = new Date(currentBlockTimestamp.getTime() + timeDifference);
+
+  return realTime;
+}
+
 const FormInput = ({
   dealProposal,
   onChange,
   accounts,
   selectedAccount,
   onSelectAccount,
   currentBlock,
+  currentBlockTimestamp,
 }: {
   dealProposal: InputFields;
   onChange: (dealProposal: InputFields) => void;
   accounts: InjectedAccountWithMeta[];
   selectedAccount: InjectedAccountWithMeta | null;
   onSelectAccount: (account: InjectedAccountWithMeta) => void;
-  currentBlock: number | null;
+  currentBlock: number;
+  currentBlockTimestamp: Date;
 }) => {
+  const startBlockRealTime = blockToTime(
+    +dealProposal.startBlock,
```

Comment:
What does this `+` do?

Reply:
It's a conversion from string to a number xd

Reply:
Number.parseInt or GTFO